plugins/http_plugin/common.hpp

```diff
@@ -0,0 +1,274 @@
+        set<string>              valid_hosts;
+
+        bool verbose_http_errors = false;
+
+        url_handlers_type  url_handlers;
+        bool                     keep_alive = false;
+    };
+
+    /**
+     * Helper type that wraps an object of type T and records its "in flight" size to
+     * http_plugin_impl::bytes_in_flight using RAII semantics
+     *
+     * @tparam T - the contained Type
+     */
+    template<typename T>
+    struct in_flight {
+        in_flight(T&& object, http_plugin_state& plugin_state)
+        :_object(std::move(object))
+        ,_plugin_state(plugin_state)
+        {
+            _count = detail::in_flight_sizeof(_object);
+            _plugin_state.bytes_in_flight += _count;
+        }
+
+        ~in_flight() {
+            if (_count) {
+                _plugin_state.bytes_in_flight -= _count;
+            }
+        }
+
+        // No copy constructor, but allow move
+        in_flight(const in_flight&) = delete;
+        in_flight(in_flight&& from)
+        :_object(std::move(from._object))
+        ,_count(from._count)
+        ,_plugin_state(std::move(from._plugin_state))
+        {
+            from._count = 0;
+        }
+
+        // No copy assignment, but allow move
+        in_flight& operator=(const in_flight&) = delete;
+        in_flight& operator=(in_flight&& from) {
+            _object = std::move(from._object);
+            _count = from._count;
+            _plugin_state = from._plugin_state;
+            from._count = 0;
+        }
+
+        /**
+         * const accessor
+         * @return const reference to the contained object
+         */
+        const T& obj() const {
+            return _object;
+        }
+
+        /**
+         * mutable accessor (can be moved from)
+         * @return mutable reference to the contained object
+         */
+        T& obj() {
+            return _object;
+        }
+
+        T _object;
+        size_t _count;
+        http_plugin_state& _plugin_state;
+    };
+
+    /**
+     * convenient wrapper to make an in_flight<T>
+     */
+    template<typename T>
+    auto make_in_flight(T&& object, http_plugin_state& plugin_state) {
+        return std::make_shared<in_flight<T>>(std::forward<T>(object), plugin_state);
+    }
+
+    /**
+     * Construct a lambda appropriate for url_response_callback that will
+     * JSON-stringify the provided response
+     *
+     * @param thread_pool - pointer to the thread pool 
+     * @param plugin_state = reference to plugin state object
+     * @param session_ptr - pointer to the session object on which to invoke send_response
+     * @return lambda suitable for url_response_callback
+     */
+    auto make_http_response_handler(std::shared_ptr<eosio::chain::named_thread_pool> thread_pool, std::shared_ptr<http_plugin_state> plugin_state, detail::abstract_conn_ptr session_ptr) {
+    return [thread_pool, plugin_state, session_ptr]( int code, std::optional<fc::variant> response )
```

Comment:
These can all be moved.
`return [thread_pool{std::move(thread_pool)}, plugin_state{std::move(plugin_state)}, session_ptr{std::move(session_ptr)}]( int code, std::optional<fc::variant> response )`